osu.Game/Online/HubClientConnector.cs

```diff
@@ -64,20 +66,28 @@ public HubClientConnector(string clientName, string endpoint, IAPIProvider api,
             this.preferMessagePack = preferMessagePack;
 
             apiState.BindTo(api.State);
-            apiState.BindValueChanged(state =>
+            apiState.BindValueChanged(state => connectIfPossible(), true);
+        }
+
+        public void Reconnect()
+        {
+            Logger.Log($"{clientName} reconnecting...", LoggingTarget.Network);
+            Task.Run(connectIfPossible);
```

Comment:
Bit strange to have this run `connectIfPossible` which sometimes... calls `disconnect()`? Is there a reason why this is the case, rather than doing something like
```csharp
if (apiState.Value <= APIState.Failing)
    return;

Logger.Log($"{clientName} reconnecting...", LoggingTarget.Network);
Task.Run(connect);
```

Reply:
Not sure, just felt having the same code used in all places was preferred.